### ISSUE TITLE
Suggestions for specific use-case

### ISSUE THREAD

**Issue by Mogady** [UKPLab/sentence-transformers#1144]
Hi, I have a project where I try to match a text with a topic and a set of keywords define this topic for example "text about specific dish", "{cooking, dinner, chicken, ....}"

one of the ideas I tried was to concatenate the keywords in one sentence and use Sentence-transformers models to match both, 
I have tried almost everything, different losses {constrative, cosine, negative ranking}.
I don't have enough dataset with scores I only have like 300 examples in shape {text, topic}, so used the cross-encoder method as an augmenter, but still none of the models I try pass the 0.5 threshold. 

so does text similarity actually is a good choice for this task? I favorited it because it is fast and I need to use this on a huge scale.
any tips to make this work? i always try fine-tuning with warm-steps, max 3 epochs.

**Comment by nreimers** [comment]
300 examples can be quite small, depending on you number of topics. How many examples are there for example for 'chicken'? Learning good representations usually takes quite some number of examples.

First, I would treat the topics individually, i.e., instead of "text", "{cooking, dinner, chicken}" I would formulate the task as [("text", "cooking"), ("text", "dinner"), ("text", "chicken")]

MultipleNegativeRankingLoss should be a good choice, but you must ensure that the neither the same text nor the same topic appears twice in a mini-batch => you need a custom mini dataloader like this one: https://github.com/UKPLab/sentence-transformers/blob/master/examples/training/paraphrases/MultiDatasetDataLoader.py


Otherwise you can also try zero-shot classification using an NLI CrossEncoder:
https://huggingface.co/cross-encoder/nli-deberta-base?candidateLabels=dinner%2C+lunch%2C+breakfast%2C+chicken%2C+beef%2C+fish%2C+vegetarian&multiClass=true&text=Fried+chicken+is+my+absolute+favorite+Southern+dish.+Sure%2C+it%E2%80%99s+simple+%E2%80%93+it%E2%80%99s+basically+just+chicken+coated+in+batter+and+then+deep-fried+until+golden+brown.+Yet+somehow%2C+it+tastes+like+heaven%21

You define topics as labels and check which are true.

If this works well, you can create using this a larger dataset which you use for training.

**Comment by Mogady** [comment]
I already used ZSH for labeling another data (12K sample) the problem is it is most of the time overconfident so most of the scores are >0.95 for the correct label I will try the MultipleNegativeRankingLoss with it again and try modifying the sampling strategy for pos and neg examples. 

Thanks for the tips
